skimage/transform/registration.py

```diff
@@ -0,0 +1,168 @@
+        If it's a sequence, each number is the number of bins for that image.
+
+    Returns
+    -------
+    cnmi : float
+        The negative of the normalized mutual information between ``image0``
+        and ``image1``.
+    """
+    return -compare_nmi(image0, image1, bins=bins)
+
+
+def register_affine(reference, target, *, cost=cost_nmi, nlevels=None,
+                    multichannel=False, inverse=True,
+                    iter_callback=lambda x: None):
+    """
+    Returns a matrix which registers the target image to the reference image
+
+    Will only return an affine transformation matrix.
+
+    Parameters
+    ----------
+    reference : ndimage
+        A reference image to compare against the target
+    target : ndimage
+        Our target for registration. Transforming this image using the
+        return value `matrix` aligns it with the reference.
+    cost : function, optional
+        A cost function which takes two images and returns a score which is
+        at a minimum when images are aligned. Uses the mean square error as
+        default.
+    nlevels : integer, optional
+        Change the maximum height we use for creating the Gaussian pyramid.
+        By default we take a guess based on the resolution of the image,
+        as extremely low resolution images may hinder registration.
+    multichannel : bool, optional
+        Whether the last axis of the image is to be interpreted as multiple
+        channels or another spatial dimension. By default, this is False.
+    inverse : bool, optional
+        Whether to return the inverse transform, which converts coordinates
+        in the reference space to coordinates in the target space. For
+        technical reasons, this is the transform expected by
+        ``scipy.ndimage.affine_transform`` to map the target image to the
+        reference space.
+    iter_callback : callable, optional
+        If given, this function is called once per pyramid level with a tuple
+        containing the current downsampled image, transformation matrix, and
+        cost as the argument. This is useful for debugging or for plotting
+        intermediate results during the iterative processs.
+
+    Returns
+    -------
+    matrix : array
+        A transformation matrix used to obtain a new image.
+        ndi.affine_transform(target, matrix) will align your target image.
+
+    Example
+    -------
+    >>> from skimage.data import camera
+    >>> reference_image = camera()
+    >>> r = 0.42  # radians
+    >>> c, s = np.cos(r), np.sin(r)
+    >>> matrix_transform = np.array([[c, -s, 0], [s, c, 50], [0, 0, 1]])
+    >>> target_image = ndi.affine_transform(reference_image, matrix_transform)
+    >>> output_matrix = register_affine(reference_image, target_image)
+    >>> registered_target = ndi.affine_transform(target_image, output_matrix)
+
+    """
+
+    ndim = reference.ndim
+
+    if nlevels is None:
```

Comment:
not sure about this determining levels, usually you set the nimilam image size and then define pyramid with the power of 2